chore: run python tests in virtual env

DanielMSchmidt opened the pull request:
Installing packages requires a non-root environment

mutahhir commented on test/python/provider-add-command/test.ts:
```diff
@@ -49,9 +48,14 @@ describe.skip("provider add command", () => {
         await driver.setupPythonProject({
           init: { additionalOptions: "--cdktf-version 0.10.4" },
         });
+        driver.setEnv("PIPENV_VERBOSITY", "-1");
```
Minor/Nit: Leaving a comment here would be useful to recall why we did this.

```
Supress warning that Pipenv is running within a virtual environment
```

mutahhir reviewed (approved):
:) Mostly don't get what's going on, but I trust you

DanielMSchmidt commented:
Just for context: We updated to a newer Python version, which marks the global install location for pip as system managed, throwing an error if against it without a special flag. For our use-cases and the ones of our customers they will most likely will use a virtual env, which we now need to set up in our tests. We updated the python version because the latest (1.93.0) version of JSII dropped support for python 3.7.
